elastic/beats
# Add build tags for cloud tests

## What does this PR do?

Add build tags to AWS and Azure integration tests, and make tests fail if they cannot get a working configuration.

Add support for a `TEST_TAGS` environment variable in mage targets that add additional build tags to `go tests` calls.
When `TEST_TAGS` includes a cloud provider, it is expected that valid credentials to run these tests are available in the environment, if not, tests will fail. This is intentional, to avoid skipping tests by mistake due to some misconfiguration in CI.

## Why is it important?

To add support of integration testing of cloud features in CI we need some way to select what tests to run in a way that can be easily defined in jenkins configuration files. With `TEST_TAGS` we can do this selection of tests.

## Checklist

- [x] ~~I have made corresponding changes to the documentation~~ (We need to review documenation about running tests in general, we also need to review how to document mage targets and supported environment variables)

## How to test this PR locally

Run integration tests for cloud features with `go test`:
* Without corresponding tags, tests shouldn't be run.
* With `go test --tags=integration,aws`, AWS integration tests should be run, with credentials they should work, without credentials they should fail.

Repeat the steps with `mage goIntegTest`:
* Without `TEST_TAGS`, cloud tests shouldn't be run.
* With `TEST_TAGS=aws`, AWS integration tests should be run, with credentials they should work, without credentials they should fail.

## Related issues

- Part of elastic/beats#15891
